conda_forge_tick/update_upstream_versions.py

```diff
@@ -23,6 +24,23 @@
 logger = logging.getLogger("conda-forge-tick._update_versions")
 
 
+def version_match(version, new_version: str, payload):
```

Comment:
You might choose a more descriptive name. Also why does this need the existing version?

Reply:
to check if the current version (after the match) corresponds to something greater the the current one

Reply:
Isn't `version_match` good ?

Reply:
I'm not certain if that is needed, since we already have the maximal version, and we check the new version against the current version in the migrator itself.

To some extent I'd rather have this logic on the migrator itself in the filter method. That way we always have the latest new version in case third parties want it.

Reply:
I would have something more verbose like `check_version_matches_cadence`.
With `version_match` what are you matching the version to?
With `check_pattern` what are you checking the pattern against, which pattern?

Reply:
Looking througly at the code you sent before (the `get_version` function) I think the it's easier just to add the `re.match(patter, version)` there. -- If almost the same operation occuring there (this is just an extension)

Reply:
Would it be possible to have it in the version migrator code?

Reply:
For what I read until now (line 400~) It can be inserted ther (inside the `VersionOrder` of filter)

Reply:
There is a possibility to insert after the 
```python
# if no new version do nothing
if "new_version" not in attrs or not attrs["new_version"]:
    return True
```

Reply:
I would put it lower, right before the final return, so that we check that we can update the version and that the version is later than the current version.

Reply:
> I would have something more verbose like `check_version_matches_cadence`.
> With `version_match` what are you matching the version to?
> With `check_pattern` what are you checking the pattern against, which pattern?

Thanks, this realy helps me a lot (not only with this but further names choices too)

Reply:
> I'm not certain if that is needed, since we already have the maximal version, and we check the new version against the current version in the migrator itself.
> 
> To some extent I'd rather have this logic on the migrator itself in the filter method. That way we always have the latest new version in case third parties want it.

This is an excellent point (storing the actual versions strings), I will do that (can I set this PR as closed then ?)

Reply:
> I would put it lower, right before the final return, so that we check that we can update the version and that the version is later than the current version.

Uhmmm, I do agree if this (it's the best logical way) but I can't understand the ` version_filter` object (more especificaly how it's been used)

Reply:
`version_filter` is a boolean that states if the feedstock should be filtered out. It has two main pieces (potentially 3 if you add your code to it) the first [clause](https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/version.py#L411) checks if the current version is greater than the new version we have. The second [clause](https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/version.py#L415) determines if we've issued a PR for a later version. If either of these clauses are true we don't issue a PR.

Reply:
BTW, why `or not attrs.get("new_version"),  # if no new version` is performed as an option? Don't we exclude it when 
```python
if "new_version" not in attrs or not attrs["new_version"]:
            return True
```

Reply:
> `version_filter` is a boolean that states if the feedstock should be filtered out. It has two main pieces (potentially 3 if you add your code to it) the first [clause](https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/version.py#L411) checks if the current version is greater than the new version we have. The second [clause](https://github.com/regro/cf-scripts/blob/master/conda_forge_tick/migrators/version.py#L415) determines if we've issued a PR for a later version. If either of these clauses are true we don't issue a PR.

It was very explicative, thank you very much 😄